CMakeLists.txt

```diff
@@ -34,6 +39,43 @@ if(CMAKE_C_COMPILER_VERSION VERSION_GREATER 9.1)
 	target_compile_options(rimage PRIVATE -Wno-char-subscripts)
 endif()
 
+if(${CMAKE_HOST_WIN32})
+	cmake_minimum_required(VERSION 3.20.0)
+	if(NOT OPENSSL_PATH)
+		message(FATAL_ERROR "Please set OPENSSL_PATH to your OpenSSL install directory")
+	endif()
+	cmake_path(ABSOLUTE_PATH OPENSSL_PATH NORMALIZE OUTPUT_VARIABLE OPENSSL_PATH)
+	cmake_path(APPEND OPENSSL_PATH "include" OUTPUT_VARIABLE OPENSSL_INCLUDE_PATH)
+	cmake_path(APPEND OPENSSL_PATH "lib" OUTPUT_VARIABLE OPENSSL_LIB_PATH)
+	cmake_path(APPEND OPENSSL_PATH "bin" OUTPUT_VARIABLE OPENSSL_BIN_PATH)
+
+	# If you link with OpenSSL .DLLs, then you're expected to include into
+	# your application code small "shim" snippet, which provides glue between
+	# OpenSSL BIO layer and your compiler run-time. See the OPENSSL_Applink
+	# manual page for further details.
+	target_include_directories(rimage PRIVATE "${OPENSSL_INCLUDE_PATH}")
+	target_link_directories(rimage PRIVATE "${OPENSSL_LIB_PATH}")
+	add_library(crypto SHARED IMPORTED GLOBAL)
+
+	set_property(TARGET crypto PROPERTY IMPORTED_IMPLIB_RELEASE
+		"${OPENSSL_LIB_PATH}/libcrypto.lib")
+	set_property(TARGET crypto PROPERTY IMPORTED_LOCATION_RELEASE
+		"${OPENSSL_BIN_PATH}/libcrypto-1_1-x64.dll")
+	set_property(TARGET crypto PROPERTY IMPORTED_IMPLIB_DEBUG
+		"${OPENSSL_LIB_PATH}/libcrypto.lib")
+	set_property(TARGET crypto PROPERTY IMPORTED_LOCATION_DEBUG
+		"${OPENSSL_BIN_PATH}/libcrypto-1_1-x64.dll")
+
+	target_sources(rimage PRIVATE "${OPENSSL_INCLUDE_PATH}/openssl/applink.c")
+	set_source_files_properties("${OPENSSL_INCLUDE_PATH}/openssl/applink.c" PROPERTIES
+		COMPILE_OPTIONS "-Wno-missing-prototypes")
+	target_compile_options(rimage PRIVATE -Wl,-EL)
+
+	add_custom_command (TARGET rimage POST_BUILD
+		COMMAND ${CMAKE_COMMAND} -E copy_if_different
+		$<TARGET_FILE:crypto> $<TARGET_FILE_DIR:rimage>)
```

Comment:
Is that enough to generate the rimage -> crypto build dependency? Apologies for the naive CMake question, you probably have more CMake knowledge than all other persons currently contributing to rimage combined.